workers/createRecurringEvents.js

```diff
@@ -90,6 +85,13 @@ module.exports = (cron, fetch) => {
                         startTime: filteredEvents[i].startTime && newEventDate,
                         endTime: filteredEvents[i].endTime && newEndTime,
                         hours: filteredEvents[i].hours && filteredEvents[i].hours
+                    } 
+                    if (filteredEvents.hasOwnProperty("location")) {
+                        eventToCreate.location = {
+                            city: filteredEvents[i].location.city && filteredEvents[i].location.city,
+                            state: filteredEvents[i].location.state && filteredEvents[i].location.state,
+                            country: filteredEvents[i].location.country && filteredEvents[i].location.country
+                        };
```

Comment:
I see that we are using a for loop using a counter. This seems like a good temporary fix. Have you thought of converting this loop to a for of loop instead? 

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

Reply:
Yep, that's a good suggestion..would help with readability. :)

Reply:
I have a quick questions for my own learning. Does each field have to have the double value of VALUE && VALUE. Maybe I am misunderstanding, but why is that needed?

Reply:
After doing a little research, I see that the value on the right is used when I try adding two string objects in the shell (see my screenshot). For readability, can we make the values that are repeated the single value

i.e. 
hacknight: filteredEvent.hacknight
vs 
hacknight: filteredEvent.hacknight && filteredEvent.hacknight,

Then you may need to review the date values listed there and try and figure out which one you want to list. It might be a turnary value of one that exists OR something else. 

![image](https://user-images.githubusercontent.com/13891656/91604519-a5064480-e923-11ea-843a-d19ab1733ab3.png)

Reply:
This is a good point - I believe the idea is following the pattern to test the property first to see if it exists before using it's value. But in this case, it defeats the purpose because whether the result is true or false, the property is still being created..

This can definitely be simplified, but probably something we should break out and address as a code refactor at some later point..the sole purpose of this fix is to address that there is a double reach into a property that is causing a exception error if no location is provided at all.

Reply:
If we were to use this pattern correctly, I think the idea would be something like  if (obj.hasOwnProperty("someProp") { newObj.someProp = obj.someProp; }
This would only add the property of someProp with it's value if it exists. 
The current method of checking during the assignment will result in a value like "false" or undefined for that property. 
Thinking about it some more, maybe this **should** be addressed with this fix.. but I would want to have a discussion on the data model as it is and it's lack of "required" properties first.

Reply:
That makes sense. Okay yeah, I think that this link might be helpful in looking at validation of schema. https://stackoverflow.com/a/38616988/5900471

For the time being, if this has the desired outcome, then I will approve this PR. I think that Alex will want to review and merge as well.

Reply:
Sounds good - yeah I would prefer PRs to add fixes or features should only address one item at a time, I feel. I'm good with whatever Alex decides on this. :)